### PR TITLE
fix(status):fix bug that stageStatus not change while aborting record

### LINKED ISSUE #444: abort record stageStatus not change
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:
- abort a record record status change to Aborted, but stageStatus not change
`{
            "id":"5ad7336a52c2b20001c76234",
            "name":"v555",
            "pipelineID":"5ad7090852c2b20001c76228",
            "trigger":"admin",
            "performParams":Object{...},
            "stageStatus":{
                "codeCheckout":{
                    "status":"Running",
                    "startTime":"2018-04-18T20:00:53.656+08:00",
                    "endTime":"0001-01-01T00:00:00Z",
                    "commits":{
                        "mainRepo":null
                    }
                }
            },
            "status":"Aborted",
            "startTime":"2018-04-18T20:00:49.423+08:00",
            "endTime":"2018-04-18T20:00:56.566+08:00"
        }`
**What you expected to happen**:
- change stageStatus
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
/priority p1
/assign @supereagle

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #444 

**Special notes for your reviewer**:

/cc @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/router/pipelineRecord.go [hunk 1]
```diff
@@ -160,17 +161,39 @@ func (router *router) updatePipelineRecordStatus(request *restful.Request, respo
...
+	// event and pipeline record both will be updated in method "UpdateEvent"
+	err = event.UpdateEvent(e)
+	if err != nil {
+		log.Errorf("update event %s error: %v", e.ID, err)
+	}
+
+	response.WriteHeaderAndEntity(http.StatusOK, pipelineRecord)
+}
+
+func updatePipelineRecordStatusFromRunningToAborted(p *api.PipelineRecord) {
```
Comment (supereagle) [inline pkg/server/router/pipelineRecord.go]: This method name is too looooong. What about `abortPipelineRecord`?

Comment (zhujian7) [reply pkg/server/router/pipelineRecord.go]: ok

File: pkg/server/router/pipelineRecord.go [hunk 2]
```diff
@@ -160,17 +161,39 @@ func (router *router) updatePipelineRecordStatus(request *restful.Request, respo
...
-		if err != nil {
-			log.Errorf("update event %s error: %v", e.ID, err)
+	updatePipelineRecordStatusFromRunningToAborted(e.PipelineRecord)
+	// event and pipeline record both will be updated in method "UpdateEvent"
+	err = event.UpdateEvent(e)
+	if err != nil {
+		log.Errorf("update event %s error: %v", e.ID, err)
+	}
+
+	response.WriteHeaderAndEntity(http.StatusOK, pipelineRecord)
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/server/router/pipelineRecord.go]:
```
// event and pipeline record both will be updated in method "UpdateEvent"
err = event.UpdateEvent(e)
......
```

[the remark this answers is not in the document]
Comment (supereagle) [reply pkg/server/router/pipelineRecord.go]: LGTM

**Comment by supereagle** [comment]
Why the coverage decreased so much (-5.0%)?

**Comment by zhujian7** [comment]
@supereagle , i am really confused with the coverage, i don't know the reason, so i decide to add some other test cases.

File: pkg/server/router/pipeline_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		},
+		"end error": {
+			"1521345720",
+			"abc",
+		},
+	}
+
+	for k, v := range d {
+		s, e, err := checkAndTransTimes(v.start, v.end)
+		fmt.Printf("%v -- %v", s, e)
```
Comment (supereagle) [inline pkg/server/router/pipeline_test.go]: Remove?

File: pkg/server/router/pipeline_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		},
+		"end = start": {
+			"1521345720",
+			"1521345720",
+		},
+	}
+
+	for k, v := range d {
+		s, e, err := checkAndTransTimes(v.start, v.end)
+		fmt.Printf("%v -- %v", s, e)
```
Comment (supereagle) [inline pkg/server/router/pipeline_test.go]: Remove this line. Output these info in `t.Errorf()` when test cases fail.

File: pkg/worker/stage/stage_test.go [hunk 5]
```diff
@@ -341,3 +344,65 @@ func TestExecIntegrationTest(t *testing.T) {
...
+	clientInterface := mock_docker.NewMockClientInterface(ctl)
+	clientInterface.EXPECT().PushImage(gomock.Any(), gomock.Any()).Return(nil)
+
+	// Log to standard error instead of files.
+	flag.Set("logtostderr", "true")
+
+	// Init the common clients.
+	endpoint := "unix:///var/run/docker.sock"
+	if runtime.GOOS == "darwin" {
+		//endpoint = "unix:////Users/robin/Library/Containers/com.docker.docker/Data/s60"
```
Comment (supereagle) [inline pkg/worker/stage/stage_test.go]: This test case can run Mac?

Comment (zhujian7) [reply pkg/worker/stage/stage_test.go]: yes, i mocked the docker client.

File: pkg/server/router/pipeline_test.go (hunk 3 above)
Comment (zhujian7) [reply pkg/server/router/pipeline_test.go]: done

File: pkg/server/router/pipeline_test.go (hunk 4 above)
Comment (zhujian7) [reply pkg/server/router/pipeline_test.go]: done

**Pull request merged**